File: src/main/kotlin/com/elissamc/core/Party.java
package com.elissamc.core;

import cn.nukkit.Player;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

/**
 * Class used for party.
 *
 * @author Mato Kormuth
 *
 */
public class Party implements PlayerHolder {
    private final List<Player> players;
    private final Player       owner;

    /**
     * Creates new party.
     *
     * @param owner
     *            owner of the party
     */
    public Party(final Player owner) {
        this.players = new ArrayList<Player>();
        this.owner = owner;

        owner.sendMessage("Created new party!");
    }

    /**
     * Returns whether is specified player owner of the party.
     *
     * @param player
     *            specified player
     * @return true or false
     */
    public boolean isOwner(final Player player) {
        return this.owner == player;
    }

    /**
     * Adds player to party.
     *
     * @param player
     *            player to be added
     */
    public void addPlayer(final Player player) {
        this.players.add(player);
        player.sendMessage("You have joined "
                + this.owner.getDisplayName() + "'s party!");
    }

    @Override
    public Collection<Player> getPlayers() {
        return this.players;
    }

    @Override
    public int getPlayerCount() {
        return this.players.size();
    }

    public void removePlayer(final Player player) {
        this.players.remove(player);
        StorageEngine.getProfile(player.getUniqueId()).setParty(null);
    }

    @Override
    public boolean contains(final Player player) {
        return this.players.contains(player);
    }
}

File: src/main/kotlin/com/elissamc/core/Party.kt
package com.elissamc.core

import cn.nukkit.Player
import com.elissamc.chat.ChatFormat

import java.util.ArrayList

/** Class used for party. */
class Party
/**
 * Creates new party.
 *
 * @param owner
 * owner of the party
 */
(private val owner: Player) : PlayerHolder {
    override val players: MutableList<Player>

    override val playerCount: Int
        get() = this.players.size

    init {
        this.players = ArrayList()

        owner.sendMessage(ChatFormat.success("Created new party!"))
    }

    /**
     * Returns whether is specified player owner of the party.
     *
     * @param player
     * specified player
     * @return true or false
     */
    fun isOwner(player: Player): Boolean {
        return this.owner === player
    }

    /**
     * Adds player to party.
     *
     * @param player
     * player to be added
     */
    fun addPlayer(player: Player) {
        this.players.add(player)
        player.sendMessage(ChatFormat.success("You have joined " + this.owner.displayName + "'s party!"))
    }

    fun removePlayer(player: Player) {
        this.players.remove(player)
        StorageEngine.getProfile(player.uniqueId).party = null
    }

    override fun contains(player: Player): Boolean {
        return this.players.contains(player)
    }
}